### ISSUE TITLE
Typo with put shorthand causes improper functionality

### ISSUE THREAD

**Issue by foxnewsnetwork** [samselikoff/ember-cli-mirage#874]
Background
===
Consider the this snippet from the update function https://github.com/samselikoff/ember-cli-mirage/blob/master/addon/orm/model.js#L75:
```javascript
    if (typeof key === 'object') {
      attrs = key;
    } else {
      (attrs = {})[key] = val;
    }
```

You'll notice that we must either call `myModel.update('attrs', newAttrs)` or `myModel.update({ attrs: newAttrs })`. However, if you inspect the short-hand for the `this.put` method at https://github.com/samselikoff/ember-cli-mirage/blob/master/addon/route-handlers/shorthands/put.js#L24, you'll notice we're using our internal api incorrectly:

```javascript
return modelClass.find(id).update(attrs);
```
Which forces the downstream user to change implementation related fields of the model instance instead of (what I assume is the desired functionality) of allowing the user to change the data stored in the `attrs` field.

Suggestion
===
Change the line at https://github.com/samselikoff/ember-cli-mirage/blob/master/addon/route-handlers/shorthands/put.js#L24
from 
```javascript
return modelClass.find(id).update(attrs);
```
to
```javascript
return modelClass.find(id).update({attrs});
```
and write some tests (lol)

If this is ok, I can do a PR

**Comment by samselikoff** [comment]
Hm, the two uses of `update` are currently

```js
user.update('firstName', 'Sam')
```

and

```js
user.update({
  firstName: 'Sam',
  lastName: 'Selikoff'
});
```

and never `myModel.update({ attrs: newAttrs })` like in your second example. Actually `model.attrs` should be considered private, and you should use `update` to change that data (foreign keys are stored on `.attrs` but they could affect things elsewhere).

The snippet

```js
if (typeof key === 'object') {
  attrs = key;
} else {
  (attrs = {})[key] = val;
}
```

is just a convenience for the implementation of that method. It says if you pass in an object, `attrs` is that object. If you pass in a string/value, make `attrs` a new object with one key, the string/value you passed in.

Does that all make sense?

**Comment by foxnewsnetwork** [comment]
> does that all make sense

Yes, it does (and thanks for the detailed explanation).

Actually, that's how I initially imagined the the `model.update` method used in `this.patch` shorthand to work. Specifically in my use case here:

https://github.com/foxnewsnetwork/ember-with-redux/blob/master/tests/dummy/mirage/config.js#L32

I imagine I would be able to just use

```javascript
this.patch('/dogs/:id'); // calls `return modelClass.find(id).update(attrs);`
```

However, for whatever reason, the `update` calls weren't going through and I was forced to manually do:

```javascript
this.patch('/dogs/:id', function(db, request) {
    const { dogs } = db;
    const id = request.params.id;
    const attrs = this.normalizedRequestAttrs();
    const dog = dogs.find(id);
    const updatedDog = dog.update('attrs', attrs);
    return updatedDog;
  });
}
```

I will look into this more to confirm I'm not insane, but if it's not too much trouble for you and I'm just making some obvious filthy-casual-tier mistakes using the api, please do point out what I should fix. Thanks

**Comment by samselikoff** [comment]
Seems right to me... you on latest version?

**Comment by foxnewsnetwork** [comment]
Yes, I'm on `0.2.1`
![screen shot 2016-09-16 at 10 01 24 am](https://cloud.githubusercontent.com/assets/1010423/18594392/a2d95ea4-7bf4-11e6-9870-26b1239aec0a.png)

**Comment by samselikoff** [comment]
Sorry, I missed your code earlier.

```js
dog.update('attrs', attrs);
```

shouldn't be used - this says you're updating the `attrs` property on the dog to some new object. What you want is

```
dog.update(attrs)
```

which updates each property, i.e.

```
dog.update({ firstName: 'Spot', lastName: 'Smith' })
```

and so on. Does that make sense? 

Also, the shorthand should work basically in this way, so if it's not we should investigate. Any chance you can reproduce in a twiddle? I have a boilerplate here: https://ember-twiddle.com/eedfd390d8394d54d5bfd0ed988a5d0f

**Comment by foxnewsnetwork** [comment]
I tried to reproduce in the twiddle here https://ember-twiddle.com/45788c18ed18037e3c7c22f438265251?openFiles=controllers.index.js%2C

But had no luck in reproducing my bug (incidentally, once I updated to mirage 0.2.2 to take advantage of traits, this bug also went away on its own). I will try looking into my original repo to see if I can't isolate what was wrong there, but in the mean-time, I think it might be safe to close this issue.

Thanks so much for your time and effort in helping me with this, btw.
